tests/test_examples.py

```diff
@@ -1293,3 +1295,177 @@ def test_optimize_with_fourier_planar_coil():
+        (
+            PlasmaCoilsetMinDistance(
+                eq=eq,
+                coils=coils,
+                target=offset,
+                weight=2,
+                plasma_grid=plasma_grid,
+                coil_grid=coil_grid,
+                eq_fixed=True,
+                coils_fixed=False,
+            ),
+            CoilsetMinDistance(
+                coils,
+                target=2 * np.pi * (R0 - offset) / coils.num_coils,
+                grid=coil_grid,
+            ),
+        )
+    )
+    # only 2 free optimization variables are coil center position phi and radius r_n
+    constraints = (
+        FixCoilCurrent(coils),
+        FixParameters(coils, {"center": np.array([0, 2]), "normal": True}),
+    )
+    optimizer = Optimizer("scipy-trf")
+    [coils_opt], _ = optimizer.optimize(
+        things=coils, objective=objective, constraints=constraints, verbose=2, copy=True
+    )
+
+    assert coils_opt[0].current == 1e6  # current was fixed
+    np.testing.assert_allclose(  # check coils are equally spaced in toroidal angle phi
+        coils_opt[0].center,
+        [R0, np.pi / coils_opt.num_coils, 0],
+        rtol=1e-5,
+    )
+    np.testing.assert_allclose(coils_opt[0].normal, [0, 1, 0])  # normal was fixed
+    np.testing.assert_allclose(
+        coils_opt[0].r_n, a + offset, rtol=1e-2
+    )  # check coil radius
+
+    #### optimize coils with fixed surface ####
+    # same optimization as above, but with a fixed surface instead of an equilibrium
+    objective = ObjectiveFunction(
+        (
+            PlasmaCoilsetMinDistance(
+                eq=surf,
+                coils=coils,
+                target=offset,
+                weight=2,
+                plasma_grid=plasma_grid,
+                coil_grid=coil_grid,
+                eq_fixed=True,
+                coils_fixed=False,
+            ),
+            CoilsetMinDistance(
+                coils,
+                target=2 * np.pi * (R0 - offset) / coils.num_coils,
+                grid=coil_grid,
+            ),
+        )
+    )
+    # only 2 free optimization variables are coil center position phi and radius r_n
+    constraints = (
+        FixCoilCurrent(coils),
+        FixParameters(coils, {"center": np.array([0, 2]), "normal": True}),
+    )
+    optimizer = Optimizer("scipy-trf")
+    [coils_opt], _ = optimizer.optimize(
+        things=coils, objective=objective, constraints=constraints, verbose=2, copy=True
+    )
+
+    assert coils_opt[0].current == 1e6  # current was fixed
+    np.testing.assert_allclose(  # check coils are equally spaced in toroidal angle phi
+        coils_opt[0].center,
+        [R0, np.pi / coils_opt.num_coils, 0],
+        rtol=1e-5,
+    )
+    np.testing.assert_allclose(coils_opt[0].normal, [0, 1, 0])  # normal was fixed
+    np.testing.assert_allclose(
+        coils_opt[0].r_n, a + offset, rtol=1e-2
+    )  # check coil radius
+
+    #### optimize surface with fixed coils ####
```

Comment:
@dpanici I modified this test to include a linear constraint that the surface cross-section should remain circular, since the coilset is circular and before it was forcing the coils to become elliptical. Also I removed the 4th test you had because it was attempting to do an optimization with both the eq and coils, but it was actually fixing the coils. It would be nice to include a realistic test with both being optimized but I can't think of a simple/easy test case.

Reply:
The 4th test was just to test the logic when both the eq and the coils were in `things`. I know that the coil `r_n` was fixed, that was intentional. It was to check the logic in the objective function

Reply:
That logic is already tested by `test_plasma_coil_min_distance` in `test_objective_functions.py`. We only need to check that `things` and `params_1` vs `params_2` get assigned in the correct order, which only requires testing that the objective can be computed without error and not a full optimization test. As you can see on [Codecov](https://app.codecov.io/github/PlasmaControl/DESC/commit/45d338af61673fe77c3b712177b2da7950870ee6), adding this test back did not change the coverage at all. I'm not trying to be annoying, I'm just trying to prevent us from adding redundant tests.

Reply:
Thank you for catching this, sorry for my hardheadedness